Assets/Character/Scripts/GroundChecker.cs

```diff
@@ -0,0 +1,44 @@
+﻿using UnityEngine;
```

Comment:
Instead of doing this create a stateless class that takes in the Collider2D
public class GroundChecker
{
    public bool OnGround(Collider2D)
}

and then just call it from your other components. There is no reason to check onGround every time we do a fixed update (from this classes perspective). Only calculate if we're on the ground when our client classes (jump.cs, roll.cs) ask for it.